src/core/components/files-regular/add-from-url.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const { URL } = require('url')
+const fetch = require('../../runtime/fetch-nodejs')
+
+module.exports = (self) => {
+  return async (url, options, callback) => {
```

Comment:
async methods that take callbacks. The async/await refactor cannot come soon enough.